### ISSUE TITLE
[a11y] Banner link color and WCAG 2.1 link without underline contrast with text colour

### ISSUE THREAD

**Issue by alex-page** [Shopify/polaris-react#1620]
## Issue summary

We have changed the link colour to `#006FBB` so that it meets a 4.5 colour contrast ratio on all background colours. We changed the link colour so that the darkest background colour  would pass, the darkest background is the critical banner `#FBEAE5`.

Changing the colour to `#006FBB` also had a negative affect that means we fail WCAG 2.1. Our text colour `#212B36` needs to meet a 3.0 contrast on our link colour without links being underlined. It now fails 3.0 contrast does not and requires all links to have underlines.

This is also strongly related to https://github.com/Shopify/polaris-ux/issues/147

## Expected behavior

To resolve this, all links in the banner component need to be underlined and monochrome. This will make our links pass contrast on the banner and allow us to slightly increase the lightness making the link colour to `#0075C0`. This will make sure that the link blue passes on the new darkest background and it meets a 3.0 contrast against the text colour.

## Actual behavior

Links are not meeting WCAG 2.1 https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

**Comment by dpersing** [comment]
> all links in the banner component need to be underlined and monochrome

Works for me!

Defining link colors and patterns (a consistent underline, for example) will be crucial in any future color/branding work.

**Comment by francinen** [comment]
I have time to take this on if there are no objections 🤚

**Comment by alex-page** [comment]
Hey @francinen that would be great, for this PR I would mainly focus on the link colour in the banner. The actual updating of the link colour to the other blue will happen separately to this work.

**Comment by francinen** [comment]
👍 Thanks for clarifying

**Comment by francinen** [comment]
Is this issue referring only to the secondary action? Do we also want to implement this change to the primary action if it is a link?

**Comment by alex-page** [comment]
I didn't do a good job at explaining:
> To resolve this, all links in the banner component need to be underlined and monochrome.

All blue links in the content area should act like the secondary link. They should be grey with an underline:

<img width="809" alt="Screen Shot 2019-06-07 at 1 29 32 PM" src="https://user-images.githubusercontent.com/19199063/59122260-52102e00-8928-11e9-8f26-3cd823763f30.png">

```jsx
<Banner
  title="High risk of fraud detected"
  action={{content: 'Review risk analysis'}}
  secondaryAction={{content: 'Review risk analysis'}}
  status="critical"
>
  <p>
    Before fulfilling this order or capturing payment, please{' '}
    <Link url="">review the Risk Analysis</Link> and determine if this order is
    fraudulent.
  </p>
</Banner>
```

**Comment by mirualves** [comment]
It seems that we only replaced the `default link` to 'monochrome link` on our banners and left behind the `external link`. Was that on purpose?

<img width="993" alt="Screenshot 2019-07-11 13 02 37" src="https://user-images.githubusercontent.com/5349023/61070563-0a743a80-a3dd-11e9-9521-c353499fd437.png">

**Comment by mirualves** [comment]
Oh, maybe I misunderstood... are the changes on this issue only affecting banners that will be developed after this issue got merged? Does it affect banners that already existed in the back of office (admin)?

<img width="977" alt="Screenshot 2019-07-11 13 13 32" src="https://user-images.githubusercontent.com/5349023/61070894-b584f400-a3dd-11e9-908c-fd83534faa24.png">

**Comment by francinen** [comment]
@mirualves Hello! The PR that resolved this issue was merged into the `version-4.0.0` feature branch, which hasn't been rolled out yet. The new behaviour will be available in Polaris v4.
